octopoes/tests/test_bit_spf_discovery.py

```diff
@@ -54,3 +54,14 @@ def test_spf_discovery_intermediate_success():
     results = list(run(dnstxt_record, [], {}))
 
     assert len(results) == 12
+
+
+def test_spf_discovery_with_identifier():
+    dnstxt_record = DNSTXTRecord(
+        hostname=Reference.from_str("Hostname|internet|example1.com"),
+        value="v=spf1 a:example.com mx mx:deferrals.domain.com ptr:otherdomain.com "
+        "exists:%i.example.com ?include:example2.com ~all",
+    )
+    results = list(run(dnstxt_record, [], {}))
+
+    assert len(results) == 10
```

Comment:
I think it would be good to check that we have "example.com" in the results and not something else.

Reply:
We remove the whole exists mechanism, so example.com should not be there. Thats why there's 10 instead of the test above which has 12.